src/GrpAb/GrpAbFinGen.jl

```diff
@@ -175,6 +175,17 @@ function show(io::IO, A::GrpAbFinGen)
   end
 end
 
+function show(io::IO, ::MIME"text/plain", A::GrpAbFinGen)
+  @show_name(io, A)
+  @show_special(io, A)
+
+  if is_snf(A)
+    show_snf(io, MIME"text/plain"(), A)
+  else
+    show_gen(io, MIME"text/plain"(), A)
+  end
+end
+
 function show_hom(io::IO, G)#::GrpAbFinGen)
```

Comment:
There are several specialized `show` functions like this one. But I'm actually not sure how/when they are used?

Reply:
Maybe if you create `direct_product([G, G])` for the direct product one?

Reply:
Okay, I don't know how, but yes.
Anyway, right now this would look like:
```
julia> direct_product(abelian_group([0, 2]), abelian_group([0, 2]))
(direct product of (Finitely generated abelian group with 2 generators and 2 relations, Finitely generated abelian group with 2 generators and 2 relations), GrpAbFinGenMap[Map: direct product of (Finitely generated abelian group, Finitely generated abelian group) -> finitely generated abelian group, Map: direct product of (Finitely generated abelian group, Finitely generated abelian group) -> finitely generated abelian group])
```
So I'll try to work out some adjustments for these.